go-controller/pkg/ovn/namespace.go

```diff
@@ -596,7 +596,7 @@ func (oc *Controller) createNamespaceAddrSetAllPods(ns string) (addressset.Addre
 	} else {
 		ips = make([]net.IP, 0, len(existingPods))
 		for _, pod := range existingPods {
-			if !pod.Spec.HostNetwork {
+			if !pod.Spec.HostNetwork && !util.PodCompleted(pod) {
```

Comment:
please tell me if I am being paranoid here:
same case as below:
podA is created, podA does its business, it goes to completed state. Update Handler will get the event and try to remove its resources (deleteLogicalPort), now let's say that failed mid-way we are now retrying the deletion of completed pod and then we upgrade (oops at a bad timing!), now new masters come up and namespace handlers start before the pod handlers, for a brief time when the pod hasn't actually being cleanup should it be part of a namespace_AS?

Reply:
I guess I am being a paranoid now..the whole thing is so rare, just ignore my question.. should be 0 probability of happening.

Reply:
@trozet : can you also exclude notScheduled pods from this list?
I don't know why but it seems like as soon as we start namespace watcher, we get a bunch of pods which are not scheduled and we up with `no pod IPs found` error messages:

```
2022-04-27T22:52:34Z {} W0427 22:50:22.455078       1 namespace.go:645] pod openshift-operator-lifecycle-manager/olm-operator-67b79749bf-pkwmt: no pod IPs found
2022-04-27T22:52:34Z {} I0427 22:50:22.455085       1 namespace.go:222] [openshift-cloud-controller-manager-operator] adding namespace took 13.313983ms
2022-04-27T22:52:34Z {} W0427 22:50:22.455096       1 namespace.go:645] pod openshift-operator-lifecycle-manager/package-server-manager-85dcfddf6b-48mxx: no pod IPs found
2022-04-27T22:52:34Z {} I0427 22:50:22.455098       1 namespace.go:218] [openshift-apiserver-operator] adding namespace
2022-04-27T22:52:34Z {} W0427 22:50:22.455102       1 namespace.go:645] pod openshift-operator-lifecycle-manager/catalog-operator-69b88df49-v7rt7: no pod IPs found
2022-04-27T22:52:34Z {} W0427 22:50:22.455107       1 namespace.go:645] pod openshift-operator-lifecycle-manager/collect-profiles-27518325-hfx99: no pod IPs found
2022-04-27T22:52:34Z {} I0427 22:50:22.455125       1 transact.go:41] Configuring OVN: [{Op:insert Table:Address_Set Row:map[external_ids:{GoMap:map[name:openshift-operator-lifecycle-manager_v4]} name:a13781166516790073860] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996192}]
2022-04-27T22:52:34Z {} W0427 22:50:22.456874       1 namespace.go:645] pod openshift-marketplace/marketplace-operator-59ffd47c54-grcjm: no pod IPs found
2022-04-27T22:52:34Z {} I0427 22:50:22.456887       1 namespace.go:222] [openshift-operator-lifecycle-manager] adding namespace took 15.129189ms
2022-04-27T22:52:34Z {} I0427 22:50:22.456897       1 namespace.go:218] [kube-public] adding namespace
2022-04-27T22:52:34Z {} I0427 22:50:22.456923       1 transact.go:41] Configuring OVN: [{Op:insert Table:Address_Set Row:map[external_ids:{GoMap:map[name:openshift-marketplace_v4]} name:a10951847637192197900] Rows:[] Columns:[] Mutations:[] Timeout:<nil> Where:[] Until: Durable:<nil> Comment:<nil> Lock:<nil> UUIDName:u2596996193}]
2022-04-27T22:52:34Z {} I0427 22:50:22.458737       1 namespace.go:222] [openshift-marketplace] adding namespace took 16.95469ms
2022-04-27T22:52:34Z {} W0427 22:50:22.458741       1 namespace.go:645] pod openshift-kube-scheduler-operator/openshift-kube-scheduler-operator-857f8d5497-dfxlv: no pod IPs found
```

Reply:
oh good point... will add that on the next iteration after reviews

Reply:
done